test/powershell/Modules/Microsoft.PowerShell.Utility/Invoke-Item.Tests.ps1

```diff
@@ -59,6 +59,31 @@ Describe "Invoke-Item basic tests" -Tags "CI" {
         }
         Get-Content $redirectFile -Raw | Should Match "usage: ping"
     }
+
+    It "Should invoke a folder without error" {
+        if ($IsWindows)
+        {
+            $shell = New-Object -ComObject "Shell.Application"
+            $windows = $shell.Windows()
+
+            $before = $windows.Count
+            Invoke-Item -Path $PSHOME
+            # give it time to open
+            Start-Sleep -Milliseconds 500
+            $after = $windows.Count
+
+            $before + 1 | Should Be $after
+            $item = $windows.Item($after - 1)
+            $item.LocationURL | Should Match ($PSHOME -replace '\\', '/')
+            ## close the windows explorer
+            $item.Quit()
+        }
+        else
+        {
+            # can't validate that it actually opened, but no error should be returned
+            Invoke-Item -Path $PSHOME
```

Comment:
lsof (w/o parameters) should show directories too - in Unix directory is file.

>It's probably because the GUI is only enumerating 

I think GUI open the directory, enumerate files and close directory. Then we run lsof...

Another idea is to change a directory default app, ex. script which create a flag file. I did this in ' "Describe "Invoke-Item tests on Windows" ' - maybe we can do the same on Unix.